### ISSUE TITLE
[Feature]: Request for Ascend NPU support

### ISSUE THREAD

**Issue by xuedinge233** [vllm-project/vllm#6368]
### 🚀 The feature, motivation and pitch

# Background

Currently, the project supports various hardware accelerators such as GPUs, but there is no support for NPUs. Adding NPU support could significationly benefit users who have access to these devices, enabling faster and more efficient computations. 

# Reference Materials

Ascend is a full-stack AI computing infrastructure for industry applications and services based on Huawei Ascend processors and software. For more information about Ascend, see [Ascend Community](https://www.hiascend.com/en/).

[CANN](https://www.hiascend.com/en/software/cann) (Compute Architecture of Neural Networks), developped by Huawei, is a heterogeneous computing architecture for AI.

Pytorch has officially announced [support for Ascend NPU](https://pytorch.org/blog/pytorch-2-1/) (through key PrivateUse1), please see the PrivateUse1 tutorial [here](https://pytorch.org/tutorials/advanced/privateuseone.html).

# Specific Request
we would like to request the addition of support for NPUs within the project. In order to achieve this goal, we will contributing code   and providing feedback. This request may additional resources and effort on your part, but we hope to get your help if possible.

**Comment by mgoin** [comment]
Duplicate of https://github.com/vllm-project/vllm/issues/1606 and https://github.com/vllm-project/vllm/issues/6066

**Comment by xuedinge233** [comment]
> Duplicate of #1606 and #6066

Thinks for your reply, we are ready to develop Ascend ADAPTS for the vllm-project, after completion, we will provide you with a public Ascend prototype

**Comment by mgoin** [comment]
That's great to hear @xuedinge233! Having hardware instances available for CI would be great for maintaining support

**Comment by xuedinge233** [comment]
Yeah, we‘re working on it, and we will upload it as soon as there is progress in the development

**Comment by BrightXiaoHan** [comment]
> Yeah, we‘re working on it, and we will upload it as soon as there is progress in the development

Any update about this?

**Comment by xuedinge233** [comment]
> > Yeah, we‘re working on it, and we will upload it as soon as there is progress in the development
> 
> Any update about this?

We're doing it, but it's not the main force right now，following the completion of the current assignment, we intend to prioritize this task. It will become primary focus.

**Comment by zer0py2c** [comment]
Is it planned to support NPU+CANN environments where the CPU is the aarch64 architecture?

**Comment by dogeeelin** [comment]
Looking forward to NPU support on vLLM！

**Comment by MengqingCao** [comment]
> Is it planned to support NPU+CANN environments where the CPU is the aarch64 architecture?

Yes, NPU+MindIE+CANN is in plan, and both aarch64 and x86_64 will be supported.

**Comment by ccly1996** [comment]
> > > Yeah, we‘re working on it, and we will upload it as soon as there is progress in the development
> > 
> > 
> > Any update about this?
> 
> We are making steady progress and expect to deliver the first version of the code in a couple of weeks.
Excuse me, is there any progress?

**Comment by shilei4260** [comment]
我看华为已经支持vllm了但是没用放出来https://support.huaweicloud.com/bestpractice-modelarts/modelarts_10_3172.html

**Comment by shilei4260** [comment]
大概还有多久放出来哦vllm,期待好久了。

**Comment by shilei4260** [comment]
我看华为已经支持vllm了但是没有放出来
https://support.huaweicloud.com/bestpractice-modelarts/modelarts_10_3172.html

**Comment by zer0py2c** [comment]
> > Is it planned to support NPU+CANN environments where the CPU is the aarch64 architecture?
> 
> Yes, NPU+MindIE+CANN is in plan, and both aarch64 and x86_64 will be supported.

I mean NPU+VLLM+CANN, and based on aarch64 CPU architecture. NPU device is Atlas 300I Duo or others.

**Comment by shilei4260** [comment]
How much longer until the vllm version of NPU is released

**Comment by shilei4260** [comment]
@xuedinge233 How much longer until the vllm version of NPU is released

**Comment by shilei4260** [comment]
@xuedinge233 
How long will it take to release the vllm version of npu

**Comment by xuedinge233** [comment]
> @xuedinge233 How long will it take to release the vllm version of npu


The latest updates on the progress can be found here https://github.com/vllm-project/vllm/pull/8054
